_posts/2022-02-04-shard-indexing-backpressure-in-opensearch.markdown

```diff
@@ -0,0 +1,205 @@
+---
+layout: post
+title:  "Shard Indexing Backpressure in OpenSearch"
+authors:
+  - sisurab
+  - sdharms
+  - tkharban
+  - ramaran
+date:   2022-02-04
+categories:
+  - feature
+---
+
+We are excited with the release of Shard Indexing Backpressure feature in OpenSearch version 1.2. In this post we dive into the feature and how it can improve cluster reliability. The Indexing APIs in OpenSearch such as *_bulk* allow~~s~~ you to write data in the cluster, which is distributed across multiple shards, on multiple data nodes. However, at times indexing requests may suffer performance degradation due to a number of reasons including non-optimal cluster configuration, shard strategy, traffic spikes, available node resources and more. These issues are further exacerbated for larger multi-node cluster and indices with many shards. All of these could cause out-of-memory errors, long GC pauses, and reduced throughput, affecting the overall availability of data nodes in addition to degrading performance. This in turn would impact the node's ability to perform useful work. In addition, these node drop events could cascades due to a lack of effective backpressure which puts the entire cluster at the risk.
+
+Indexing Backpressure offers a real-time solution to address such issues, particularly for multi-node clusters with large data sets that have high throughput requirements. Indexing backpressure offers selective load-shedding of indexing requests when performance thresholds are exceeded. For example, in the case of too many stuck requests or request in cluster running too long, indexing backpressure would kick in. This not only protects clusters from these types of cascading failures, but also ensures that cluster can continue to perform as expected by only isolating the impacted paths. By tracking real-time performance on per shard basis in the background and evaluating them against dynamic thresholds, Indexing Backpressure applies appropriate in-time rejections decision at the shard-request level whenever a shard or a node suffers a duress situation affecting its ability to keep up with the other shards/nodes in the cluster. This prevents wide failures while also guaranteeing fairness in request execution.
```

Comment:
* 'request in cluster running too long' -> 'request in the cluster running too long'
* 'ensures that cluster can continue' -> 'ensures that the cluster can continue'